codegen/gql_code_builder/lib/src/inline_fragment_classes.dart

```diff
@@ -94,28 +97,34 @@ List<Spec> buildInlineFragmentClasses({
         // print("alias $typeName => ${dataClassAliasMap[typeName]!.symbol}");
         return false;
       }
+      // is it okay to inlcude interfaces?
       return true;
     }).expand(
       (inlineFragment) => buildSelectionSetDataClasses(
-          name: "${name}__as${inlineFragment.typeCondition!.on.name.value}",
-          selections: mergeSelections(
-            [
-              ...selections.whereType<FieldNode>(),
-              ...selections.whereType<FragmentSpreadNode>(),
-              ...inlineFragment.selectionSet.selections,
-            ],
-            fragmentMap,
-          ),
-          fragmentMap: fragmentMap,
-          dataClassAliasMap: dataClassAliasMap,
-          schemaSource: schemaSource,
-          type: inlineFragment.typeCondition!.on.name.value,
-          typeOverrides: typeOverrides,
-          superclassSelections: {
-            name: SourceSelections(url: null, selections: selections)
-          },
-          built: built,
-          whenExtensionConfig: whenExtensionConfig),
+        name: "${name}__as${inlineFragment.typeCondition!.on.name.value}",
+        selections: mergeSelections(
+          [
+            ...selections.whereType<FieldNode>(),
+            ...selections.whereType<FragmentSpreadNode>(),
+            ...inlineFragment.selectionSet.selections,
+          ],
+          fragmentMap,
+        ),
+        fragmentMap: fragmentMap,
+        possibleTypesMap: possibleTypesMap,
+        dataClassAliasMap: dataClassAliasMap,
+        schemaSource: schemaSource,
+        type: inlineFragment.typeCondition!.on.name.value,
+        typeOverrides: typeOverrides,
+        superclassSelections: {
+          name: SourceSelections(url: null, selections: selections),
+          ...Map.fromEntries(superclassSelections.entries.map((e) => MapEntry(
+              "${e.key}__as${inlineFragment.typeCondition!.on.name.value}",
+              e.value))),
+        },
```

Comment:
@knaeckeKami  I think the cause is that this fix about `buildInlineFragmentClasses` is not checking feature flag.
This is made to cover more interfaces like adding `GheroFieldsFragment__asHuman` interface to `GheroFieldsFragmentData__asHuman`. And I think changes made by this are all additions for missing interfaces but no regressions of existing interfaces. So there would be no regression issue even without the feature flag.

Reply:
I just run the code generation without the flag and unfortunately, it produces code that does not compile.

the following graphql document caues problems:

```graphql
query HeroWithInterfaceSubTypedFragments($episode: Episode!) {
  hero(episode: $episode) {
    ...heroFieldsFragment
  }
}

fragment heroFieldsFragment on Character {
  id
  name

  ...on Human {
    ...humanFieldsFragment
  }

  ...on Droid {
    ...droidFieldsFragment
  }
}

fragment humanFieldsFragment on Human {
  homePlanet
  friends {
    ...on Droid {
      id
      name
      ...droidFieldsFragment
    }
    ...on Human {
      id
      name
      homePlanet
    }
  }
}

fragment droidFieldsFragment on Droid {
  primaryFunction
}

```

this adds another superclass to GheroFieldsFragmentData__asHuman which conflicts with the implementation:


```
'GheroFieldsFragmentData__asHuman.friends' ('BuiltList<GheroFieldsFragmentData__asHuman_friends?>? Function()') isn't a valid override of 'GheroFieldsFragment__asHuman.friends' ('BuiltList<GheroFieldsFragment__asHuman_friends?>? Function()'). (Documentation) 
The member being overridden (hero_with_interface_subtyped_fragments.data.gql.dart:151).
```

Reply:
@knaeckeKami sorry to bother you, I will turn off new logics when there is no feature flag. Will share again soon.

Reply:
no worries! i really appreciate you working on this! :)

Reply:
@knaeckeKami
How are you doing well? Finally came back here.
Sorry for late response.  Just made the update. 😀